app/controllers/api/base_controller.rb

```diff
@@ -37,6 +37,24 @@ def verify_mfa_requirement
     render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
   end
 
+  def response_with_warnings(response)
+    response_with_mfa_warning(response)
+  end
+
+  def response_with_mfa_warning(response)
+    message = response
+    if @api_key.user.mfa_recommended_not_yet_enabled?
```

Comment:
Is there any way that we know for sure that `response_with_mfa_warning` will only be called after `@api_key` is set? 

I see we're setting it as part of the before action in `DeletionsController` where we intend to use this, but as this is a method on BaseController, it seems this has a hidden dependency on the `authenticate_with_api_key` method being called before `response_with_mfa_warning`. 

Should we handle nil `@api_key` here so that it doesn't blow up if this method is used on its own?